internal/download_client/connection.go

```diff
@@ -258,3 +262,20 @@ func (s *service) testReadarrConnection(ctx context.Context, client domain.Downl
 
 	return nil
 }
+
+func (s *service) testPorlaConnection(client domain.DownloadClient) error {
+	p := porla.NewClient(porla.Settings{
+		Hostname:  client.Host,
+		AuthToken: client.Settings.APIKey,
+	})
+
+	version, err := p.Version()
+
+	if err != nil {
+		return errors.Wrap(err, "porla: failed to get version: %v", client.Host)
+	}
+
+	s.log.Debug().Msgf("test client connection for porla: found version %s (commit %s)", version.Version, version.Commitish[:8])
```

Comment:
not that it's devastating in Go (or the product because this is on Test), but if the length is less than 8 and you're reading past it's an error.